stingray/base.py

```diff
@@ -2134,6 +2134,147 @@ def sort(self, reverse=False, inplace=False):
             mask = mask[::-1]
         return self.apply_mask(mask, inplace=inplace)
 
+    def fill_bad_time_intervals(
+        self,
+        max_length=None,
+        attrs_to_randomize=None,
+        buffer_size=100,
+        uniform=None,
+        seed=None,
+    ):
+        """Fill short bad time intervals with random data.
+
+        To fill the gaps in all but the time points (i.e., flux measures, energies), we take the
+        ``buffer_size`` (default 100) valid data points closest to the gap and repeat them randomly
+        with the same empirical statistical distribution. That is, if the "blabla" attributes, in
+        the 100 points of the buffer, has 30 times 10, 10 times 9, and 60 times 11, there will be
+        *on average* 30% of 10, 60% of 11, and 10% of 9 in the simulated data.
+
+        Times are treated differently depending on the fact that the time series is uniformly
+        sampled or not. If it is not, the times are simulated from a uniform distribution with the
+        same count rate found in the buffer. Otherwise, times just follow the same grid used
+        inside GTIs. Using the uniformly sampled or not is decided based on the ``uniform``
+        parameter. If left to ``None``, the time series is considered uniformly sampled if
+        ``self.dt`` is greater than zero and the median separation between subsequent times is
+        within 1% of the time resolution.
```

Comment:
I think there are some important limitations here that we might need to point out either in the docstring or in whatever demo notebook this is used in. 

Limitation 1: I'm pretty sure this will generate white noise? If so, this might have important effects on timing properties. Related: you're not actually picking from the distribution, but from the 100 points nearest to the gap. 100 points is not that much for sampling with repetition, especially if the gap is much longer than 100 data points, so that could also have significant effects on the analysis done with the data. A somewhat more involved version of this would involve fitting a function to the CDF and then picking from that CDF, but that might be overkill here.

Limitation 2: unevenly sampled light curves are almost *never* sampled following a Poisson distribution (drawing from a uniform distribution corresponds to a Poisson distribution of arrivals with constant rate parameter), so the sampling will probably be significantly different between the GTIs and bad time intervals. Side note: the use of "uniform" here is a bit confusing, because you talk about "uniform" sampling, i.e. sampled evenly in time, and drawing non-uniformly sampled data from a uniform distribution. Might need disambiguation?

Reply:
Yes, the motivation here is that the gaps to be filled need to be _really_ short. I'm talking about, e.g., the half-second gaps often popping up in NICER data. In these conditions, the statistical distribution of the simulated data is negligible. I should probably point this out more clearly. I will change the wording to eliminate at least some of the confusion

Reply:
And for uniform... I just had some piece of my brain screwed wrong that day and couldn't come up with the word "evenly sampled" :D
I changed the wording, now it should be less confusing

Reply:
Also, I fixed the buffer. Now, by default, it cannot be shorter than the estimated number of samples in `max_length`.